core/domain/story_services.py

```diff
@@ -613,3 +617,247 @@ def record_completed_node_in_story_context(user_id, story_id, node_id):
     if node_id not in progress_model.completed_node_ids:
         progress_model.completed_node_ids.append(node_id)
         progress_model.put()
+
+
+def get_story_rights_from_model(story_rights_model):
+    """Constructs a StoryRights object from the given story rights model.
+
+    Args:
+        story_rights_model: StoryRightsModel. Story rights from the
+            datastore.
+
+    Returns:
+        StoryRights. The rights object created from the model.
+    """
+
+    return story_domain.StoryRights(
+        story_rights_model.id,
+        story_rights_model.manager_ids,
+        story_rights_model.story_is_published
+    )
+
+
+def publish_story(story_id, committer_id):
+    """Marks the given story as published.
+
+    Args:
+        story_id: str. The id of the given story.
+        committer_id: str. ID of the committer.
+
+    Raises:
+        Exception. The given story does not exist.
+        Exception. The story is already published.
+        Exception. The user does not have enough rights to publish the story.
+    """
+    def _is_node_valid_for_publishing(node):
+        if not node.exploration_id:
+            raise Exception(
+                'Story node does not contain an exploration id.')
+        if not exp_services.get_exploration_by_id(
+                node.exploration_id, strict=False):
+            raise Exception(
+                'Exploration id %s doesn\'t exist.' % node.exploration_id)
+        exploration_rights = rights_manager.get_exploration_rights(
+            node.exploration_id, strict=False)
+        if exploration_rights.is_private():
+            raise Exception(
+                'Exploration with id %s isn\'t published.'
+                % node.exploration_id)
+
+    story = get_story_by_id(story_id, strict=False)
+    for node in story.story_contents.nodes:
+        if node.id == story.story_contents.initial_node_id:
+            _is_node_valid_for_publishing(node)
+
+    story_rights = get_story_rights(story_id, strict=False)
```

Comment:
Consider doing these checks before retrieving the nodes; it's a less costly operation and it's better to fail fast before the more expensive computation.

Reply:
Yes right! Done.